### ISSUE TITLE
Error code: MODIFICATION_FAILED

### ISSUE THREAD

**Issue by dilom** [rkokkelk/todo.txt-ext#7]
An error occurred when writing to the calendar Todo.txt!

This happens when I click the checkbox to show the Todo.txt calendar. 

Any ideas?

**Comment by rkokkelk** [comment]
Okay, 

Have you specified the correct location of your todo.txt and done.txt file? If that is not the case could you specify some further details so I can resolve this issue as soon as possible. 

Thank you for your feedback.

**Comment by dilom** [comment]
No, I did not...bleh. I totally missed the Browse button where you specify the location of the files. I was adding several plugins and each had a pop up window for preferences, so I missed the settings for this one completely. My bad.

I also have a question about the implementation. I have one main calendar file and when I enter a new task it does not automatically go to Todo.txt (which is added as a separate calendar in Thunderbird). I have to select which calendar it should go to, otherwise it won't go into Todo.txt. I don't know if this is relevant to other users but I rather use one Calendar that synchs todo.txt. Ideally have an option of synch todo.txt to current default calendar or import todo.txt as a separate calendar (as currently).

I also had some issues after specifying the file location (it worked but had some unexpected behavior ). After restarting Thunderbird it worked as expected so I will report back only if the same issues come back again. 

Great plugin, thanks for your time!

**Comment by rkokkelk** [comment]
First of all great suggestion. I will definitely look into this because I think a lot of people have the same preference. Also in the future I would like to add additional exception handling so that it is clear that a file could not be found and show a pop-up when installing the add-on. Thanks for the feedback.

Could you elaborate on the issues surrounding first specifying the file location, because that should work right away, without a reboot. If you could give me any detail please let me know.

**Comment by HelpdeskThisIsJohn** [comment]
Same problem.  I'm using Todo.txt 0.5.  The todo.txt and done.txt files are on my local synced version of Dropbox.

Todo.txt appears twice in my calendar list. I've tried unsubscibing to one or both but they re-appear when T-bird is restarted. Trying to create a task and pressing save causes  the error.   Thunderbird 38.6.0, Lignthening 4.0.6

**Comment by rkokkelk** [comment]
Sorry for my late reply, but I'm definitely working on this. However the first part is to alter the add-on into a restartless add-on and that requires some work. Once that is finished I have a lot more control on which calendar is used and created which should prevent this circumstances. 

I'll let you know once I publish a fix.

**Comment by HelpdeskThisIsJohn** [comment]
Roy -

Thank you.  Right now I'm stuck using Google Tasks which doesn't really 
cut it.

One thing I thought was interesting was that I removed the add-on and 
turned off T-Bird completely, rebooted.  I found I was still getting an 
error!.  A bit of testing with another GUI, and I found that there seems 
to be sharing conflict between DropBox and whatever is editing the file.

I wouldn't put too much work into the rewrite intil you have confirmed 
that its the add-on and not Todo.txt itself.

Best of luck,

John

**Comment by muralikodali** [comment]
Event **added to todo.txt on command line** is appearing well on thunderbird calendar.But when i tried to add event to calendar in thunderbird it is showing error: **MODIFICATION_FAILED**.
I have added write permission to todo.txt file.But it didn't also work.

**Comment by rkokkelk** [comment]
@muralikodali the todo.txt add-on is meant for tasks and not events. I specified that it should not handle events and that is why the error occurs. You can add, view, alter tasks in the specific sidebar or window. If you have any further issues, please create a new issues so that other participants won't receive any unnecessary notifications. Thanks

**Comment by rkokkelk** [comment]
@HelpdeskThisIsJohn  the Add-on now includes functionality which removes the todo.txt calendar when the add-on is removed. So this should fix your issues and I'm therefore closing this issue. If you any further issues feel free to reopen this issue.
